src/coreclr/src/debug/debug-pal/win/diagnosticsipc.cpp

```diff
@@ -144,11 +145,14 @@ IpcStream *IpcStream::DiagnosticsIpc::Accept(ErrorCallback callback)
     {
         if (callback != nullptr)
             callback("Failed to GetOverlappedResults for NamedPipe server", ::GetLastError());
-        return nullptr;
+        // clean up the pipe
+        ::CloseHandle(_hPipe);
```

Comment:
We should also set `_hPipe = INVALID_HANDLE_VALUE;`

Reply:
That happens on line 158 just below.